ocaml/typing/datarepr.ml

```diff
@@ -56,7 +56,7 @@ let constructor_existentials cd_args cd_res =
     match cd_res with
     | None -> []
     | Some type_ret ->
-        let arg_vars_set = free_vars (newgenty (Ttuple tyl)) in
+        let arg_vars_set = free_vars (newgenty (Ttuple (List.map (fun ty -> None, ty) tyl))) in
```

Comment:
Maybe we should give this the same treatment we gave `deep_occur`.   (But this is fine for now - just resolve this comment after reading)